packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2154,6 +2178,10 @@ Hardhat Network's forking functionality only works with blocks from at least spu
         (blockContext.header as any).baseFeePerGas = new BN(0);
       }
 
+      this._vm._common.setHardfork(
```

Comment:
Isn't this affecting the next thing that runs against `this._vm._common`? We should:
1. surround subsequent `runTx()` with a `try`/`finally` with a `finally` that re-sets the hardfork to what it was before this call execution.
2. Add a test for this.